src/cobra/io/mat.py

```diff
@@ -232,18 +342,74 @@ def create_mat_dict(model: Model) -> OrderedDict:
     return mat
 
 
+def mat_annotations(
+    target_list: List[Object], mat_struct: np.ndarray, _d_replace: str = D_MET
+) -> None:
+    """Process mat structure annotations in place.
+
+    Will process mat structured annotations and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    target_list: list[cobra.Object]
+        A list of cobra objects, including metabolites, reactions or genes. The
+        annotations will be added to these lists.
+    mat_struct: np.ndarray
+        A darray that includes the data imported from matlab file.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET (for metabolite).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    annotation_lists = [[None]] * len(annotation_matlab)
+    annotation_providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    for i in range(len(annotation_matlab)):
+        annotation_lists[i] = _cell_to_str_list(mat_struct[annotation_matlab[i]][0, 0])
+        if annotation_matlab[i] == "rxnReferences":
+            # noinspection PyUnresolvedReferences
+            annotation_lists[i] = [
+                x.replace("PMID:", "") if x else None for x in annotation_lists[i]
+            ]
+        if annotation_matlab[i] == "rxnECNumbers":
+            # if there are more than one ec code, turn them to a comma separated str
+            # noinspection PyUnresolvedReferences,PyTypeChecker
+            annotation_lists[i] = [
```

Comment:
Don't join them but keep them as lists. COBRAPY annoation value scan be lists for a single provider like `rx.annotation["ec-code"] = ["1.1.1.1", "1.2.1.1"]` for instance. Everything else won't be handled correctly by the SBML writer.

Reply:
So single values should also be in a list?